src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -38,15 +44,12 @@ public override bool IsWhiteSpace
         {
             get
             {
-                foreach (var value in _buffer)
+                if (_buffer == null)
                 {
-                    if (!string.IsNullOrWhiteSpace(value))
-                    {
-                        return false;
-                    }
+                    return true;
```

Comment:
Might be worth `<remarks>` that `IsWhiteSpace` is `true` for a `null` or cleared `DefaultTagHelperContent`. This is an existing issue related to the property name but worth fixing.